src/Assistants/DataSources/AssistantDataSource.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+namespace Invoice\Assistants\DataSources;
+
+use Invoice\Services\SettingsService;
+use Plenty\Modules\Wizard\Models\WizardData;
+use Plenty\Modules\Wizard\Services\DataSources\BaseWizardDataSource;
+
+class AssistantDataSource extends BaseWizardDataSource
+{
+    /**
+     * @var SettingsService
+     */
+    protected $settingsService;
+
+    public function __construct(
+        SettingsService $settingsService
+    )
+    {
+        $this->settingsService = $settingsService;
+    }
+
+    /**
+     * @return WizardData WizardData
+     */
+    public function findData()
+    {
+        /** @var WizardData $wizardData */
+        $wizardData = pluginApp(WizardData::class);
+        $wizardData->data = ['default' => false];
+
+        return $wizardData;
+    }
+
+    /**
+     * @return array
+     */
+    private function getEntities()
+    {
+        $data = [];
+        $pids = $this->settingsService->getInvoiceClients();
+        foreach ($pids as $pid) {
+            $settings = $this->settingsService->loadClientSettingsIfExist($pid, null);
+            if (count($settings)) {
+                $settings = $this->settingsService->getSettingsForPlentyId($pid, null);
```

Comment:
Ich verstehe nicht was hier gemacht wird. Du lädst die Settings und wenn sie gesetzt sind, lädst du sie nochmal über eine andere Methode?

Reply:
Werde die Funktion etwas bearbeiten und umbenennen, weil sie eigentlich nur da war um zu überprüfen, ob es bereits Einstellungen gibt. In getSettingsForPlentyId werden die Daten nicht nur geladen, sondern auch verarbeitet und in die richtige Form gebracht.